src/app/util/af-types.h

```diff
@@ -1252,26 +1253,91 @@ typedef void (*EmberAfDefaultResponseFunction)(chip::EndpointId endpoint, chip::
 
 namespace chip {
 /**
- * @brief a type that represents where we are trying to send a message.  This
- *        must always be paired with an EmberOutgoingMessageType that identifies
- *        which arm of the union is in use.
- */
-union MessageSendDestination
-{
-    explicit constexpr MessageSendDestination(uint8_t aBindingIndex) : mBindingIndex(aBindingIndex) {}
-    explicit constexpr MessageSendDestination(NodeId aNodeId) : mNodeId(aNodeId) {}
-    explicit constexpr MessageSendDestination(GroupId aGroupId) : mGroupId(aGroupId) {}
-    explicit constexpr MessageSendDestination(Messaging::ExchangeContext * aExchangeContext) : mExchangeContext(aExchangeContext) {}
-
-    // Used when the type is EMBER_OUTGOING_VIA_BINDING
-    uint8_t mBindingIndex;
-    // Used when the type is EMBER_OUTGOING_DIRECT
-    NodeId mNodeId;
-    // Used when the type is EMBER_OUTGOING_MULTICAST or
-    // EMBER_OUTGOING_MULTICAST_WITH_ALIAS
-    GroupId mGroupId;
-    // Used when the type is EMBER_OUTGOING_VIA_EXCHANGE
-    Messaging::ExchangeContext * mExchangeContext;
+ * @brief a type that represents where we are trying to send a message.
+ *        The variant type identifies which arm of the union is in use.
+ */
+class MessageSendDestination
+{
+public:
+    struct OutGoingBinding
```

Comment:
Every single one of these is prefixed with "OutGoing".  Why?  The original constants are, because they are in the global namespace, but there's no need to do that here.

Combined with my proposal for statics above, we should be able to do things like:

```
auto destination = MessageSendDestination::Binding(index);
```
and
```
    if (destination.IsDirect()) {
      ....
    }
```